### PR TITLE
Journal worksheets

### PR DESCRIPTION
**Pull request by jackocleebrown**
This branch adds the option to write the journal log into separate worksheets within the xls file.

### CODE REVIEW

File: test/integration/detailed_export_test.rb [hunk 1]
```diff
@@ -43,6 +43,10 @@ def test_to_export_with_watchers
   def test_to_export_with_journals
     assert_columns_options 'settings_journal'
   end
+  
+  def test_to_export_with_journals
```
Comment (two-pack) [inline test/integration/detailed_export_test.rb]: The method name is already used in this test class.

File: lib/xls_export.rb [hunk 2]
```diff
@@ -393,12 +398,17 @@ def issues_to_xls2(issues, project, query, options = {})
...
 
-      def journal_details_to_xls(issue, options)
+      def journal_details_to_xls(issue, options, book = nil)
         issue_updates = issue.journals.includes(:user, :details).order("#{Journal.table_name}.created_on ASC").to_a
         return nil if issue_updates.size == 0
 
         Spreadsheet.client_encoding = 'UTF-8'
-        book = Spreadsheet::Workbook.new
+        
+        write = false
```
Comment (two-pack) [inline lib/xls_export.rb]: I think a condition that **options[:journal_worksheets] is false** is better than the flag **write** .
It is more clear that journal books are created when the option is unchecked.

**Review by two-pack — CHANGES_REQUESTED** [review]
Thank you for requesting :)
I reviewed your request.

File: test/integration/detailed_export_test.rb (hunk 1 above)
Comment (jackocleebrown) [reply test/integration/detailed_export_test.rb]: Thank you, I'll fix!

File: lib/xls_export.rb (hunk 2 above)
Comment (jackocleebrown) [reply lib/xls_export.rb]: I agree that your idea is clearer for the reader. However, I have a concern with your suggestion detailed below.

If the caller doesn't provide any book (i.e. book=nil) but **options[:journal_worksheets]** is **false** then there will be an exception at the line **book.write(xls_stream)**.

Therefore I think the code I wrote is a little more rhobust: if no book is provided then we always create one irrespective of the options. Please let me know your comments or suggestions.

Comment (two-pack) [reply lib/xls_export.rb]: I understand your concern.
I commit 4c1db73 and try to remove **write** flag and changed parameter name of **book**.
I think it's better than write flag.
Please let me know your comments or suggestions.

Comment (jackocleebrown) [reply lib/xls_export.rb]: Thank you, your code is very clear and it removes my concern. I think that your ruby is a lot better than mine too!

**Review by two-pack — APPROVED** [review]

**Comment by two-pack** [comment]
I added Japanese translation and merged it.
Thank you for pull requesting your work :)

**Comment by jackocleebrown** [comment]
Thank you for merging, and thanks for maintaining a very useful tool!

**Pull request merged**